repeatLandsatWidths.js

```diff
@@ -0,0 +1,250 @@
+////////////////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////////////////
+// Load GRWL cross sections:
+//var fc = ee.FeatureCollection('ft:1pG_qSXq5HsKizBuK3RlKg2RRxV3KlPi1WMMfjTbu'); // GRWL centerlines >120 m
+//var fc = ee.FeatureCollection('ft:14JtAPbJ9ZiSvTWKzU7SlBx2BYP0HbnidBmKD1thr'); // Xsec 200 per seg
+//var fc = ee.FeatureCollection('ft:1qcwRoPanK9dw3xTjuBqYuuxlvVPAzRFkqne6V87L'); // Xsec 20 per seg
+//var fc = ee.FeatureCollection('ft:1g-_P5jF4dJVrceGYUaNcCZF-wgZoKdZIg7i5OWF2'); // Xsec 2 examples
+var fc = ee.FeatureCollection('ft:16S4bhTENwMkDVbp3jNOmCUUgwiy-8MsTCGgvke9I');// 40 cross sections per seg 
+//var fcProperties = fc.getInfo().features[0].properties;
+//print('Cross Section FC feature properties:', fcProperties);
+
+// buffer feature and edit attributes: 
+var buffered = fc.map(function(f) {
+  f = f.set({xSecLength: f.geometry().length().toLong()}); // add feature length attribute
+  f = f.buffer(15); // buffer feature geometry
+  return ee.Feature(f.select(["segmentID", "segmentInd", "xSecLength"], null, true)); // remove needed attributes
+});
+
+print("buffered feature collection:", buffered);
+
+// get bounds of a feature collection:
+//var fcBounds = ee.Filter.bounds(buffered);
+//var fcBounds = buffered.bounds();
+//print("test", fcBounds);
+
+
+
+
+
+
+
+
+
+////////////////////////////////////////////////////////////////////////
+// Load Landsat collections:
+// fixme: consider pansharpening for 15 m resolution:
+
+// Match common names to the sensor-specific bands:
+var LT5_BANDS = ['B1',   'B2',    'B3',  'B4',  'B5',    'B7',    'B6'];
+var LE7_BANDS = ['B1',   'B2',    'B3',  'B4',  'B5',    'B7',    'B6_VCID_1'];
+var LC8_BANDS = ['B2',   'B3',    'B4',  'B5',  'B6',    'B7',    'B10'];
+var STD_NAMES = ['blue', 'green', 'red', 'nir', 'swir1', 'swir2', 'temp'];
+
+
+// load Landsat 5,7,8 collections:
+
+//var LT5 = ee.ImageCollection('LANDSAT/LT5_L1T_TOA')
+//    .select(LT5_BANDS, STD_NAMES);
+var LT5 = ee.ImageCollection('LANDSAT/LT5_L1T')
+    .select(LT5_BANDS, STD_NAMES); 
+// var LE7 = ee.ImageCollection('LANDSAT/LE7_L1T_TOA')
+//   .select(LE7_BANDS, STD_NAMES);
+// var LC8 = ee.ImageCollection('LANDSAT/LC8_L1T_TOA')
+//   .select(LC8_BANDS, STD_NAMES);
+
+var collection = LT5;//.merge(LE7).merge(LC8);
+
+
+
+//var collection = ee.ImageCollection('LANDSAT/LC8_L1T_TOA');
+var filtered = collection
+    //.filterMetadata('WRS_PATH', 'equals', 34)
+    //.filterMetadata('WRS_ROW', 'equals', 27)
+    .filterDate('2012-11-01', '2012-12-31')
+    .sort('system:time_start', false)
+    .filterMetadata('CLOUD_COVER', 'less_than', 20)
+    .filterBounds(fc);
+
+//console.log(filtered);
+//console.log(filtered.getInfo());
+
+
+
+
+////////////////////////////////////////////////////////////////////////
+// classify filtered collection as water or cloud+shadow, & add cloud attribute:
+var maskCollection = filtered.map(function(image){
```

Comment:
With a map function this long, instead of using an anonymous function, you might consider defining the function as a stand-alone named function with some documentation above it (i.e. input and output args), then just provide the function name in the `map()` call.  That will make the series of main processing steps more concise since you won't have all these lines of code within the map call.